### PR TITLE
fixed issue of Font Size animates on appearance change

### LINKED ISSUE #40708: Global Styles: Typography > Text > Font Size animates when appearance is selected.
### Description

Originally reported on [Trac 55626](https://core.trac.wordpress.org/ticket/55626).

There is a minor CSS bug caused by the dynamic width set on the Font size button within the Typography -> Styles section.

Suggested fix from @igorradovanov: Switch from percentage width to fixed-width (like vw or px) resolves the issue.

### Step-by-step reproduction instructions

1. Go to Styles.
2. Typography -> Text.
3. Change the Appearance from the dropdown.
4. The active font size will jump on the screen from right to left and back.

### Screenshots, screen recording, code snippet

(from @igorradovanov)
Before suggested fix: https://www.dropbox.com/s/5pwvsbn73askodt/Animation.gif?dl=0
After the suggested fix: https://www.dropbox.com/s/dgfjut05t7wq95i/Animation-2.gif?dl=0

### Environment info

(costdev):
I have tested this on:
- WordPress 5.9, 5.9.1, 5.9.2, 5.9.3, 6.0-trunk.
- Gutenberg: With and without 13.1.0 activated.
- Browser: Chrome 100.0.4896.127
- Device: Desktop with Windows 10
- Environment: WSL2 with Apache

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by akasunil**
## What?

In site editor sidebar, width of Font size button within the Typography change while appearance drop-down is open. I have looked into details. and found that scrollbar is causing this issue. when we open the dropdown it goes outside of screen vertically and get scrollbar into the picture. and with scrollbar, container div size get reduced by 16px. 

Fixes https://github.com/WordPress/gutenberg/issues/40708.
 
## Testing Instructions

1.  Go to Site Editor  
2.  Go to Styles.
3.  Typography -> Text.
4.  Change the Appearance from the dropdown.
5.  Check that font size button width is same as before.

### CODE REVIEW

**Review by ciampo — CHANGES_REQUESTED** [review]
Hey @sunil25393 , thank you for working to fix the issue described in #40708 ! Your contribution to the Gutenberg and the components package is greatly appreciated :)

Unfortunately, although effectively your changes fix that specific issue, they change the way the `Navigator` component handles overflows, causing the whole editor canvas to resize when the sidebar shows a scrollbar.

I think that the issue here is not about layout shifts, but more about the fact that the [`ToggleGroupControl` component](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-togglegroupcontrol--default) animates its backdrop when it resizes.

TL;DR: in order to fix this issue, I would look at disabling such animation when the component is being resized.

**Comment by akasunil** [comment]
Hey @ciampo, Thank you for you reply. I think Issue is not about the animation. The Cause is component-panel width that get reduced by 15px when scrollbar is active. which result in reduced width of font size controller-body to 265px from 280px.

Please check below screenshot. 

**Default view:**
CleanShot 2022-06-02 at 10 33 30@2x

**With scrollbar:**
CleanShot 2022-06-02 at 10 33 57@2x

I have checked same thing for post editor sidebar panel. if look at the below screenshot here, sidebar shift to right when scrollbar is active. that makes the width to remain same as 280px. and this issue doesn't happen there.
CleanShot 2022-06-02 at 10 55 25

Let me know if this makes any sense.

**Comment by ciampo** [comment]
Hey @sunil25393 ,

thank you for the detailed explanation!

Basically what's happening is the following:
- opening the dropdown causes the sidebar content's to overflow its container
- therefore, a scrollbar is shown
- depending on how each browser/OS handles the sidebar, the appearance of the sidebar can cause the layout to shift / change width
- when this happens, the size of the components inside of the sidebar adapts in a responsive way
- this resizing causes the backdrop element in the `ToggleGroupControl` component to adapt to its new position — the resizing of the backdrop happens via an animation, which makes it more visible (which is what is describe in https://github.com/WordPress/gutenberg/issues/40708)

I personally believe that layout shift caused by scrolling content (like in this case) are an expected behaviour/feature of the responsive web, and that #40708 should be fixed by working on the `ToggleGroupControl` component itself

**Comment by akasunil** [comment]
Hello @ciampo 

Thank you for your reply. I understand your concern. I have last few questions. We have same functionality on post sidebar. Where sidebar shift when scrollbar is active. We cant have same for **style sidebar** also ?

also, if we update/remove animation from **ToggleGroupControl**, won't it affect at every places ( also where this issue doesn't appear ) ?

**Comment by ciampo** [comment]
> We have same functionality on post sidebar. Where sidebar shift when scrollbar is active. We cant have same for style sidebar also ?

Even if the sidebars used in the post editor and in the side editor share the same wrapper, the contents are different in the two editors — in particular, the sidebar in the site editor uses the `Navigator` component, while the sidebar in the post editor doesn't at the moment.

What you point out is true — the behaviour of the 2 sidebars is slightly different when it comes to resizing after a scrollbar is made visible. At the same time:

- I still believe that the root of this issue is in the `ToggleGroupControl` component, animating its backdrop even when not necessary
- the site editor is still a relatively new feature, and the controls in the Global Styles sidebar are still marked as experimental (and therefore, I'd argue that it's expected to have these slight discrepancies across global styles sidebar in the site editor, and sidebar in the post editor)
- the changes introduced by this PR introduce a regression in the `Navigator` component, since they show the horizontal scrollbars during an animation from a screen to another (which is also flagged by the code comments just before the line where you applied your changes)

> if we update/remove animation from ToggleGroupControl, won't it affect at every places ( also where this issue doesn't appear ) ?

Mine was a suggestion on how to solve the issue (I haven't tried implementing a solution myself) — but what I'm suggesting is to remove the animation from the backdrop when it's not needed. The backdrop would still animate as expected when the selected item changes.

**Comment by akasunil** [comment]
Hello @ciampo 

Thank you for detailed explanation. I really appreciate it. Anyway, ill update my code accordingly.

**Comment by akasunil** [comment]
Don't see this issue anymore in latest pull. `ToggleGroupControl` components width remain same with/without active scrollbar. Closing this PR now.
